Makefile

```diff
@@ -0,0 +1,29 @@
+BENCH_ITER=10
+
+.PHONY: clangfmt
+clangfmt:
+
+
+.PHONY: memtests
+memtests: src/verify.c src/tests.c
+	clang -g $? -o ./$@ -Wall -Wextra -Werror -lcrypto -lefence -DEXTRA_DEBUG -DBENCH_ITER=$(BENCH_ITER)
+	valgrind --read-var-info=yes --track-origins=yes --leak-check=full ./$@
```

Comment:
nit: might be nice to have some additional sanitizers and fuzzing too:

* https://clang.llvm.org/docs/AddressSanitizer.html
* https://clang.llvm.org/docs/MemorySanitizer.html
* https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html
* fuzzing-project or afl

Reply:
neat!  I'll look into those and see if they work with the code or at least get an idea of what's required.